ts/Core/Axis/Axis3D.ts

```diff
@@ -368,7 +368,7 @@ class Axis3D {
     /**
      * @optionparent xAxis
      */
-    public static defaultOptions: Highcharts.AxisOptions = {
+    public static defaultOptions: DeepPartial<Highcharts.AxisOptions> = {
```

Comment:
Why is this? Did you tried `Highcharts.XAxisOptions` without deep partial?

On a side note: Maybe we should build Highcharts.AxisOptions as a deep partial intersect type instead of an union. Defaults should not use it though, instead their specific axis interface.

Reply:
It's the `defaultOptions` of the Axis3D component, so it shouldn't hold the full `AxisOptions`

Reply:
Wasn't the idea of the full XAxisOptions to only have the defaults non-optional? I still do not understand why defaults need to be partial. They should by definition contain all non-optionals, therefor default values and have to work without DeepPartial. Something is not working, if this is necessary.

Reply:
The `Axis3D.defaultOptions` are a partial of the full `AxisClass.defaultOptions` because they are merged in here:

https://github.com/highcharts/highcharts/blob/fa028f8596197eb0809b6524123317e9c8bb58b1/ts/Core/Axis/Axis3D.ts#L490

If not partial, we would need a separate type for the 3d axis options overrides.